### PR TITLE
Added parameter to control if files are sorted before being passed to pandoc

### PR DESCRIPTION
**Pull request by colinbull**
Small PR to allow the sorting behaviour to be controlled before the file sets are passed to the pandoc process. Previously files were always sorted. However, this can lead to the wrong ordering in the generated documents if the sorted order file names don't represent the required order in the generated documents.

### CODE REVIEW

File: pypandoc/__init__.py [hunk 1]
```diff
@@ -131,6 +133,10 @@ def convert_file(source_file:Union[list, str, Path, Generator], to:str, format:U
     :param bool sandbox: Run pandoc in pandocs own sandbox mode, limiting IO operations in readers and writers to reading the files specified on the command line. Anyone using pandoc on untrusted user input should use this option. Note: This only does something, on pandoc >= 2.15
             (Default value = False)
 
+    :param bool sort_files: causes the files to be sorted before being passed to pandoc (Default value = True)
```
[the remark this answers is not in the document]
Comment (colinbull) [reply pypandoc/__init__.py]: Good spot, should be fixed with latest commit

**Comment by JessicaTegner** [comment]
Hi @colinbull  thanks for the pr.

I added a comment, and then there's some CI failing

**Comment by colinbull** [comment]
Humm the build failing looks like python is failing to install correctly on the windows builds: 

`No Python at '"C:\hostedtoolcache\windows\Python\3.12.6\x64\python.exe'`

Don't have time at the minute but will look at this, although it is likely not related to this PR.

**Comment by JessicaTegner** [comment]
yes indeed doesn't seem related. I'll try rerunning the failing jobs later, to see if it's a GH issue, cause it seems very strange.

**Comment by colinbull** [comment]
No worries, I'll check back if they have passed then all good, if not I'll take a gander and submit a fix. Want a separate PR or happy to do it in here?

**Comment by JessicaTegner** [comment]
If it comes down to it, I would prefer a separate fix, and then we'll just merge  main back into this

**Comment by JessicaTegner** [comment]
@colinbull  it seems that the CI failing before was a fluke or something. It seems to be running now.
If you are finished with the pr. We can merge it

**Comment by colinbull** [comment]
Ahhh, great. Sorry, I actually completely forgot I said I'd look at this. 

As far as I'm concerned this is complete. So happy to merge.

**Pull request merged**